hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/ElasticByteBufferPool.java

```diff
@@ -36,7 +36,7 @@
  */
 @InterfaceAudience.Public
 @InterfaceStability.Stable
-public class ElasticByteBufferPool implements ByteBufferPool {
+public abstract class ElasticByteBufferPool implements ByteBufferPool {
```

Comment:
Hopefully making ElasticByteBufferPool.java abstract won't break anything as I see you have changed all the implementation.

Reply:
I have tested it; just replace the class, compile with no error, run the binary, and watch it works as I expected(no direct memory leakage).
But it requires a new test case. I will add some.